source/extensions/common/wasm/null/BUILD

```diff
@@ -0,0 +1,55 @@
+licenses(["notice"])  # Apache 2
+
+load(
+    "//bazel:envoy_build_system.bzl",
+    "envoy_cc_library",
+    "envoy_package",
+)
+
+envoy_package()
+
+envoy_cc_library(
+    name = "null_plugin_hdr",
+    hdrs = [
+        "proxy_wasm_externs.h",
+    ],
+    deps = [
+        "//api/wasm/cpp:intrinsics_hdr",
```

Comment:
So... We cannot do that (at least, not yet).

`"//api/wasm/cpp:intrinsics_hdr"` includes `<string_view>`, which is C++17, but Envoy is C++14, so this fails.

Reply:
Converted to using StringView which is one or the other.

Reply:
You need to add `copts = ["-DNULL_PLUGIN=1"],` to this target, otherwise it won't be picked up by `proxy_wasm_externs.h`.

Reply:
Erm, actually, you need to define it all over the place, since it's transitively included (via `null.h`) in a lot of places.

Maybe add `#define NULL_PLUGIN` (or some other define, to not abuse this one) in `null.h` before including `proxy_wasm_externs.h`?

Alternatively, maybe test for C++17 in `proxy_wasm_externs.h` using `#ifdef __cplusplus >= 201703L`?

Reply:
Now the only place that needs this is plugin.cc.  PTAL